app/services/list_languages_taught_in_modern_language_course.rb

```diff
@@ -0,0 +1,44 @@
+class ListLanguagesTaughtInModernLanguageCourse
+  LANGUAGES = [
+    'english as a foreign language',
+    'french',
+    'german',
+    'italian',
+    'japanese',
+    'mandarin',
+    'russian',
+    'spanish',
+    'urdu'
+  ].freeze
+
+  MODERN_LANGUAGE_COURSE_PATTERN = /mfl|modern language|modern foreign language/.freeze
+
+  attr_reader :course_name, :ucas_subject_names
+
+  def self.call(course_name, ucas_subject_names)
+    new(course_name, ucas_subject_names).call
+  end
+
+  def initialize(course_name, ucas_subject_names)
+    @course_name        = course_name
+    @ucas_subject_names = ucas_subject_names
+  end
+
+  def call
+    return [] unless modern_language_course?
+
+    LANGUAGES.select do |language|
+      language.in? normalized_ucas_subject_names
+    end
+  end
+
+private
+
+  def normalized_ucas_subject_names
+    @normalized_ucas_subject_names ||= ucas_subject_names.strip.downcase
+  end
+
+  def modern_language_course?
+    course_name.downcase.match?(MODERN_LANGUAGE_COURSE_PATTERN)
```

Comment:
Modern languages should be any course tagged to a language subject.

Reply:
For example:
https://find-postgraduate-teacher-training.education.gov.uk/course/2DO/3DQB (which is tagged to Languages (European), French, Languages, Secondary)

Reply:
So what you're saying is that it's not restricted to modern languages courses, but any "secondary" course?

I'm not sure how to differentiate the two, as both can have languages.

Reply:
I'm guessing `age_range = 'S'` (secondary)?

Reply:
@defong has answered my question - thanks